lib/spack/spack/graph.py

```diff
@@ -499,15 +496,50 @@ def graph_dot(specs, deptype='all', static=False, out=None):
         spack graph --dot qt | dot -Tpdf > spack-graph.pdf
 
     """
+    if not specs:
+        raise ValueError("Must provide specs ot graph_dot")
+
     if out is None:
         out = sys.stdout
     deptype = canonical_deptype(deptype)
 
+    def static_graph(spec, deptype):
+        pkg = spec.package
+        possible = pkg.possible_dependencies(
+            expand_virtuals=True, deptype=deptype)
+
+        nodes = set()  # elements are (node name, node label)
+        edges = set()  # elements are (src key, dest key)
+        for name, deps in possible.items():
+            nodes.add((name, name))
+            edges.update(set((name, d) for d in deps))
```

Comment:
Should work even without `set`:
```suggestion
            edges.update((name, d) for d in deps)
```

Reply:
fixed.